### ISSUE TITLE
Feature Gate: Update to burn revoked feature lamports

### ISSUE THREAD

**Issue by buffalojoec** [solana-labs/solana-program-library#6177]
#### Problem
As we get ready to migrate built-in (native) programs to BPF, we'll likely start with the Feature Gate program, a program that does not currently exist at `Feature111111...`. In this repository is the proposed new Feature Gate program with one single instruction: `RevokePendingActivation`.

This instruction is designed to defund the feature account of its lamports and send them to some designated destination. In order to add additional protection against spam, it may make more sense to instead burn these lamports.

#### Solution
If so desired, update the program to do so!

cc @joncinque - we should have some discussion on this beforehand, but I wanted to set a reminder here.

**Comment by joncinque** [comment]
> In order to add additional protection against spam, it may make more sense to instead burn these lamports.

I think I'm missing something, how does burning the lamports protect against spam?

**Comment by buffalojoec** [comment]
> > In order to add additional protection against spam, it may make more sense to instead burn these lamports.
> 
> 
> 
> I think I'm missing something, how does burning the lamports protect against spam?

It's not a full solution but it at least helps disincentivize people from spam-creating feature accounts.

**Comment by joncinque** [comment]
People can do that with all sorts of programs, like the stake program or spl-token. The solution is probably to create protocol-level fee for allocating accounts, but I would consider that out of scope for this issue.

**Comment by CriesofCarrots** [comment]
> People can do that with all sorts of programs, like the stake program or spl-token. The solution is probably to create protocol-level fee for allocating accounts, but I would consider that out of scope for this issue.

True, but there aren't limits per se to the number of stake accounts or spl-token accounts that can exist; there's no real malicious motivation for someone to create and then empty short-lived accounts (aside from using validator resources, etc). Whereas the feature process we are considering has a low hard limit to the number of features to be considered each epoch.

I do agree that requiring some higher deposit or fee is probably the way to go, but burning the lamports at least prevents someone from griefing us for free.

**Comment by joncinque** [comment]
Ah right, sorry, I knew I was missing something! I read back through the proposal to refresh my memory, and it says activation is gated by a multisig: https://github.com/solana-foundation/solana-improvement-documents/pull/72/files#diff-b798a8c9dca99c07b780074b5abdc2c8d81722af0a57dcdde447fbe96bbc8ad1R88-R95, which means that the multisig would need grief the system, no?

And separately, what would happen if someone just created thousands of feature accounts and left them there?

**Comment by buffalojoec** [comment]
> I read back through the proposal to refresh my memory, and it says activation is gated by a multisig

Yeah, I added this after the bunch of us spoke offline, but I think that SIMD still needs another update and more discussion.

> And separately, what would happen if someone just created thousands of feature accounts and left them there?

Garbage collection! IMO these should get cleared and lamports burnt on epoch rollover.
